lib/layouts/conversation_details/conversation_details.dart

```diff
@@ -128,21 +130,81 @@ class _ConversationDetailsState extends State<ConversationDetails> {
             SliverToBoxAdapter(
               child: readOnly
                   ? Container()
-                  : Padding(
-                      padding: const EdgeInsets.all(16.0),
-                      child: TextField(
-                        cursorColor: Theme.of(context).primaryColor,
-                        readOnly: true,
-                        controller: controller,
-                        style: Theme.of(context).textTheme.bodyText1,
-                        autofocus: false,
-                        autocorrect: false,
-                        decoration: InputDecoration(
-                          labelText: "NAME",
-                          labelStyle: TextStyle(color: Theme.of(context).primaryColor),
+                  : showNameField ?
+                    Row(
+                      children: [
+                        Expanded(
+                          child: Padding(
+                            padding: const EdgeInsets.only(left: 16.0, right: 8.0, bottom: 8.0),
+                            child: TextField(
+                              cursorColor: Theme.of(context).primaryColor,
+                              readOnly: !chat.isGroup(),
+                              onSubmitted: (String newName) async {
+                                await widget.chat.changeName(newName);
+                                widget.chat.getTitle();
+                                setState(() {
+                                  showNameField = newName.isNotEmpty;
+                                });
+                                EventDispatcher().emit("refresh", null);
+                              },
+                              controller: controller,
+                              style: Theme.of(context).textTheme.bodyText1,
+                              autofocus: false,
+                              autocorrect: false,
+                              decoration: InputDecoration(
+                                labelText: chat.displayName.isEmpty ? "SET NAME" : "NAME",
+                                labelStyle: TextStyle(color: Theme.of(context).primaryColor),
+                                enabledBorder: UnderlineInputBorder(borderSide: BorderSide(color: Colors.grey))
+                              ),
+                            ),
+                          ),
                         ),
+                        if (chat.displayName.isEmpty)
```

Comment:
Should you use the variable you created above here?

Reply:
Not here because we only want to show the "cancel" button next to the TextField if the displayName is empty (i.e. the user is adding a name so they should be able to cancel that action).